### PR TITLE
Create gift cards when fulfilling order

### PR DESCRIPTION
**Pull request by IKarbowiak**
Create gift cards when fulfilling order with gift card products, create `BOUGHT` events.
Send non-shippable gift cards to the customer, create `SENT_TO_CUSTOMER` events.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3899 Create gift cards when fulfilling gift card order](https://app.clickup.com/t/2549495/SALEOR-3899)

File: saleor/giftcard/utils.py [hunk 1]
```diff
@@ -49,3 +59,56 @@ def activate_gift_card(gift_card: GiftCard):
...
+    app: Optional["App"],
+    manager: "PluginsManager",
+):
+    """Create purchased gift cards."""
+    customer_user = order.user
+    user_email = order.user_email
+    gift_cards = []
+    non_shippable_gift_cards = []
+    for order_line in gift_card_lines:
+        price = order_line.total_price_gross
```
Comment (fowczarek) [inline saleor/giftcard/utils.py:78]: We should handle situations when customers buy 5x gift card(5USD).
Current flow skips line quantity.

File: saleor/graphql/order/mutations/fulfillments.py [hunk 2]
```diff
@@ -197,28 +203,63 @@ def clean_input(cls, data):
...
         order = cls.get_node_or_error(info, order, field="order", only_type=Order)
         data = data.get("input")
 
         cleaned_input = cls.clean_input(data)
 
-        user = info.context.user
+        context = info.context
+        user = context.user if not context.user.is_anonymous else None
+        app = context.app
+        manager = info.context.plugins
```
Comment (fowczarek) [inline saleor/graphql/order/mutations/fulfillments.py:242]:
```suggestion
        manager = context.plugins
```

**Review by fowczarek — CHANGES_REQUESTED** [review]

**Comment by IKarbowiak** [comment]
Commits on the base branch have changed. I'll open new PR for these changes.

**Comment by IKarbowiak** [comment]
New PR: https://github.com/mirumee/saleor/pull/7835

**Pull request closed without merging**